pennylane/templates/subroutines/trotter.py

```diff
@@ -438,10 +438,9 @@ def compute_decomposition(*args, **kwargs):
         ops = kwargs["base"].operands
 
         decomp = _recursive_expression(time / n, order, ops)[::-1] * n
-        unique_decomp = [copy.copy(op) for op in decomp]
 
         if qml.QueuingManager.recording():
-            for op in unique_decomp:  # apply operators in reverse order of expression
+            for op in decomp:  # apply operators in reverse order of expression
```

Comment:
I'm confused how this leads to the same instance getting queued multiple times? `qml.apply` copies ops that are already present in the queue before applying them.

Reply:
PennyLane internals do not use the queuing behaviour of `Operator.decomposition`. They strictly use the output. 

And `_recursive_expression` will return a list with the same operation instance multiple times.